request.js

```diff
@@ -991,6 +991,7 @@ Request.prototype.onRequestResponse = function (response) {
     // Hack to emit end on close because of a core bug that never fires end
     response.on('close', function () {
       if (!self._ended) {
+        self._ended = true
```

Comment:
mildly worried that just setting this flag to true will cause issues with stream flow, let me set up tests for this repo and see

Reply:
ok, I could always set another internal prop and `if` that

Reply:
all the tests pass here & in Cypress, so I suppose this is fine